pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -0,0 +1,99 @@
+package gencrypto
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// AuthConfig is an asset that generates ECDSA public and private keys.
+type AuthConfig struct {
+	PublicKey, PrivateKey string
+}
+
+var _ asset.WritableAsset = (*AuthConfig)(nil)
+
+// Dependencies returns the assets on which the AuthConfig asset depends.
+func (a *AuthConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the auth config for agent installer APIs.
+func (a *AuthConfig) Generate(dependencies asset.Parents) error {
+	PublicKey, PrivateKey, err := keyPairPEM()
+	if err != nil {
+		return err
+	}
+	a.PublicKey = PublicKey
+	a.PrivateKey = PrivateKey
+
+	return nil
+}
```

Comment:
Would be nice to have a test on the `Generate` method

Reply:
Not sure if we need a test for this here. A test is added in the ignition though.

Reply:
In general, if an asset produces something is worth to be tested... The test in the ignition is a very narrow one, just limited to the private function `getTemplateData`